File: 'highest scoring word'/high.hs
import Data.List (maximumBy)
import Data.Char (ord)
import Data.Function (on)

score :: String -> Int
score = sum . (map (\c -> (ord c) - 96))

high :: String -> String
high = maximumBy (compare `on` score) . reverse . words

main = putStrLn $ high "aa b"

File: 'highest scoring word'/high.rs

// ASSCI code a is 97. 
fn high(input: &str) -> &str {
    let score = |word: &str| -> usize {word.chars().map(|c| c as usize - 96).sum()};
    input.split_whitespace().rev().max_by_key(|word| score(word)).unwrap()
}

fn main() {
    println!("{}", high("aa b"));
}
